bson/src/main/org/bson/BasicBSONEncoder.java

```diff
@@ -281,10 +281,22 @@ protected void putDate(final String name, final Date date) {
      * Encodes any number field.
      *
      * @param name   the field name
-     * @param number the value
+     * @param number The value. The following types are supported:
```

Comment:
I needed to see a set of specific `Number` types that we automatically convert to BSON types, to document those types as supported by `SearchFacet.numberFacet`. If `BasicBSONEncoder.putNumber` is the wrong method to use as a reference, do you know what method/class we may suggest instead?

Reply:
Maybe `org.bson.codecs.ValueCodecProvider`?  The issue is that the set of supported types is not fixed, but rather based on the configuration of the codec registry.

Reply:
Rolled back the changes related to `BasicBSONEncoder`. The search API does not need any of that anymore because it now utilizes the codec system and does not need to know how to encode `Number`.